### PR TITLE
add support signet

### PR DESCRIPTION
**Pull request by wakiyamap**
From https://github.com/spesmilo/electrum/pull/7454

I want to lightning electrum with signet.

Please let me know if there is a problem.
I also think that there are not enough test items, so it would be more helpful if you could tell me where you need to add them.

Thanks.

### CODE REVIEW

File: src/main/scala/fr/acinq/bitcoin/Protocol.scala [hunk 1]
```diff
@@ -288,6 +288,7 @@ object Message extends BtcSerializer[Message] {
   val MagicMain = 0xD9B4BEF9L
   val MagicTestNet = 0xDAB5BFFAL
   val MagicTestnet3 = 0x0709110BL
+  val MagicSigNet = 0x40CF030AL
```
Comment (t-bast) [inline src/main/scala/fr/acinq/bitcoin/Protocol.scala:291]: Can you link me to where that magic value comes from?

**Review by t-bast — COMMENTED** [review]
Concept ACK, we've talked about supporting signets for a while but just haven't had time to look into it.

If I understand correctly, there can be a lot of different signets. They all share the same genesis block, but there is a way to join different signets. How does that work in practice? What is used to figure out which signet you want to join?

File: src/main/scala/fr/acinq/bitcoin/Protocol.scala (hunk 1 above)
Comment (wakiyamap) [reply src/main/scala/fr/acinq/bitcoin/Protocol.scala:291]: from rust-bitcoin
https://github.com/rust-bitcoin/rust-bitcoin/blob/0.27.0/src/network/constants.rs#L93

**Comment by wakiyamap** [comment]
It is now connected to a signet that is currently running in public.
It is not possible to connect to custom signet with the current description.

``eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala``
```
case "signet" => bitcoinClient.invoke("getrawtransaction", "c0c2e89bcf21c54e59fddfd65de16e0764ee2f6c0e93a37e5e78690e754d3310") // coinbase of #45000
```

Maybe fail here.

File: src/main/scala/fr/acinq/bitcoin/Protocol.scala (hunk 1 above)
Comment (t-bast) [reply src/main/scala/fr/acinq/bitcoin/Protocol.scala:291]: Thanks.

**Review by t-bast — APPROVED** [review]
LGTM, so for now we only support one specific signet.
In the future we can add support for connecting to other signets, we'll see when that's needed.

**Pull request merged**